src/Centreon/Domain/HostConfiguration/Model/HostCategory.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+/*
+ * Copyright 2005 - 2020 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+declare(strict_types=1);
+
+namespace Centreon\Domain\HostConfiguration\Model;
+
+use Centreon\Domain\Common\Assertion\Assertion;
+
+/**
+ * This class is designed to represent a host category.
+ *
+ * @package Centreon\Domain\HostConfiguration\Model
+ */
+class HostCategory
+{
+    public const MAX_NAME_LENGTH = 200,
+                 MAX_ALIAS_LENGTH = 200,
+                 MAX_COMMENTS_LENGTH = 65535;
+
+    /**
+     * @var int|null
+     */
+    private $id;
+
+    /**
+     * @var string Define a short name for this category. It will be displayed with this name in the ACL configuration.
+     */
+    private $name;
+
+    /**
+     * @var string Longer description of this category.
+     */
+    private $alias;
+
+    /**
+     * @var string|null Comments regarding this category.
+     */
+    private $comments;
+
+    /**
+     * @var bool Indicates whether this host category is enabled or not (TRUE by default)
+     */
+    private $isActivated = true;
+
+    /**
+     * @return int|null
+     */
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param int $id
+     * @return HostCategory
+     */
+    public function setId(int $id): HostCategory
+    {
+        $this->id = $id;
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    /**
+     * @param string $name
+     * @return HostCategory
+     * @throws \Assert\AssertionFailedException
+     */
+    public function setName(string $name): HostCategory
```

Comment:
shouldn't you check for a minLenght = 3 characters ?

Reply:
Where did you find this rule because this rule does not exist for the frontend.

Reply:
Indeed. I rechecked and can't find them